models/block.go

```diff
@@ -60,6 +78,72 @@ func decodeBlockEvent(event cadence.Event) (*Block, error) {
 			ReceiptRoot:         payload.ReceiptRoot,
 			TransactionHashRoot: payload.TransactionHashRoot,
 			TotalGasUsed:        payload.TotalGasUsed,
+			PrevRandao:          payload.PrevRandao,
 		},
 	}, nil
 }
+
+type blockV0 struct {
+	ParentBlockHash     gethCommon.Hash
+	Height              uint64
+	Timestamp           uint64
+	TotalSupply         *big.Int
+	ReceiptRoot         gethCommon.Hash
+	TransactionHashRoot gethCommon.Hash
+	TotalGasUsed        uint64
+}
+
+// Adds PrevRandao field
+type blockV1 struct {
+	ParentBlockHash     gethCommon.Hash
+	Height              uint64
+	Timestamp           uint64
+	TotalSupply         *big.Int
+	ReceiptRoot         gethCommon.Hash
+	TransactionHashRoot gethCommon.Hash
+	TotalGasUsed        uint64
+	PrevRandao          gethCommon.Hash
+}
+
+func pastBlockHash(b any) (gethCommon.Hash, error) {
+	data, err := rlp.EncodeToBytes(b)
+	return gethCrypto.Keccak256Hash(data), err
+}
+
+// decodeBlockBreakingChanges will try to decode the bytes into all
+// previous versions of block type, if it succeeds it will return the
+// migrated block, otherwise it will return nil.
+func decodeBlockBreakingChanges(encoded []byte) *Block {
+	b0 := &blockV0{}
+	if err := rlp.DecodeBytes(encoded, b0); err == nil {
+		return &Block{
+			Block: &types.Block{
+				ParentBlockHash:     b0.ParentBlockHash,
+				Height:              b0.Height,
+				Timestamp:           b0.Timestamp,
+				TotalSupply:         b0.TotalSupply,
+				ReceiptRoot:         b0.ReceiptRoot,
+				TransactionHashRoot: b0.TransactionHashRoot,
+				TotalGasUsed:        b0.TotalGasUsed,
+			},
+		}
+	}
+
+	b1 := &blockV1{}
+	if err := rlp.DecodeBytes(encoded, b1); err == nil {
+		return &Block{
+			Block: &types.Block{
+				ParentBlockHash:     b1.ParentBlockHash,
+				Height:              b1.Height,
+				Timestamp:           b1.Timestamp,
+				TotalSupply:         b1.TotalSupply,
+				ReceiptRoot:         b1.ReceiptRoot,
+				TransactionHashRoot: b1.TransactionHashRoot,
+				TotalGasUsed:        b1.TotalGasUsed,
+				PrevRandao:          b1.PrevRandao,
+			},
+		}
+	}
```

Comment:
Do we need this, given the Block itself ? is similar to BlockV1

Reply:
BlockV1 has the addition of the `PrevRandao` field, which BlockV0 does not have. Unless you mean something else which I'm missing :sweat_smile:

Reply:
Ah right, now I see what you mean :innocent:

Reply:
Removed in https://github.com/onflow/flow-evm-gateway/pull/463/commits/2f97be1585e3db414178e61a727d0c370d4dcd79